prow/cmd/branchprotector/protect.go

```diff
@@ -238,28 +239,24 @@ func (p *Protector) UpdateRepo(orgName string, repo string, repoDefaults config.
 	}
 	for _, branch := range branches {
 		bn := branch.Name
-		if err := p.UpdateBranch(orgName, repo, bn, branch.Protected); err != nil {
+		if err := p.UpdateBranch(orgName, repo, bn); err != nil {
 			return fmt.Errorf("UpdateBranch(%s, %s, %s, %t) failed: %v", orgName, repo, bn, branch.Protected, err)
 		}
 	}
 	return nil
 }
 
 // Update the branch with the specified configuration
-func (p *Protector) UpdateBranch(orgName, repo string, branchName string, protected bool) error {
+func (p *Protector) UpdateBranch(orgName, repo string, branchName string) error {
 	bp, err := p.cfg.GetBranchProtection(orgName, repo, branchName)
 	if err != nil {
 		return err
 	}
-	if bp == nil || bp.Protect == nil {
```

Comment:
This is correct

Reply:
Yeah. I figured it out. However I found another issue. In my config it was protecting repos that were not specified for prow jobs. I'll update this PR